src/symbolize/gimli.rs

```diff
@@ -122,13 +121,11 @@ impl<'data> Context<'data> {
             if cfg!(not(target_os = "aix")) {
                 let data = object.section(stash, id.name()).unwrap_or(&[]);
                 Ok(EndianSlice::new(data, Endian))
+            } else if let Some(name) = id.xcoff_name() {
+                let data = object.section(stash, name).unwrap_or(&[]);
+                Ok(EndianSlice::new(data, Endian))
             } else {
-                if let Some(name) = id.xcoff_name() {
-                    let data = object.section(stash, name).unwrap_or(&[]);
-                    Ok(EndianSlice::new(data, Endian))
-                } else {
-                    Ok(EndianSlice::new(&[], Endian))
-                }
```

Comment:
I would somewhat prefer this stay as-is because "all the AIX code in this fn is at its own indentation" is on-purpose. I didn't specifically say so it was when I accepted this code, but I picked the ["rightward drift"](https://github.com/rust-lang/backtrace-rs/pull/508#discussion_r1297899219) and other [stylistic nits](https://github.com/rust-lang/backtrace-rs/pull/508#discussion_r1297889865) [several](https://github.com/rust-lang/backtrace-rs/pull/508#discussion_r1297901492) [times](https://github.com/rust-lang/backtrace-rs/pull/508#discussion_r1297891386) and this was not an oversight.

If you would prefer refactoring this in a different way that appeases the lint and keeps the current visible divide between "AIX" and "everyone else" then that's fine too.